core/domain/fs_domain.py

```diff
@@ -371,6 +392,27 @@ def assets_path(self):
         """
         return self._assets_path
 
+    def _validate_entity_parameters(self, entity_name, entity_id):
```

Comment:
Can you find a way to not duplicate the code? Perhaps define a superclass and put it there?

Reply:
I'll create a global function in this file and use it?

Reply:
I think superclass is better (FileSystem or GeneralFileSystem), since this is more a case of common logic between two similar things, rather than two unrelated things using the same function.

You can create the superclass, put the function in there, then have EntityFileSystem and GcsFileSystem subclass it.

Speaking of which, EnttiyFileSystem really should be renamed DatastoreBackedFileSystem, but that's a separate issue I guess.

Reply:
Yeah, ok.

Reply:
Done